builtin/logical/database/backend_test.go

```diff
@@ -554,8 +727,8 @@ func TestBackend_connectionCrud(t *testing.T) {
 	}
 
 	delete(resp.Data["connection_details"].(map[string]interface{}), "name")
-	if !reflect.DeepEqual(expected, resp.Data) {
-		t.Fatalf("bad: expected:%#v\nactual:%#v\n", expected, resp.Data)
+	if diff := deep.Equal(resp.Data, expected); diff != nil {
```

Comment:
What's the criteria for switching to deep.Equal()? It's fine, though I also see new code using reflect.Equal() in this PR.

Reply:
The new code is just where I've copied and pasted a test; where I've added further conditions I've tended towards deep.Equal.

There's no real criteria, I just tend to use it for new statements.